e_methods.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Style/CaseEquality
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      result = []
+      my_each { |value| result.push(value) if yield(value) }
+    elsif is_a?(Range)
+      result = []
+      my_each { |value| result.push(value) if yield(value) }
+    else
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) }
+    end
+    result
+  end
+
+  def my_all?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return false unless element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif argum.nil?
+      my_each { |element| return false if element.nil? || element === false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return false unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return false unless argum.match(element) }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  def my_any?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true if element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return true if yield(element) == true }
+    elsif argum.nil?
+      my_each { |element| return true if element.nil? || element == false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return true if element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true if argum.match(element) }
+    elsif argum
+      my_each { |element| return true if element == argum }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
+
+  def my_none?(argum = nil)
+    if block_given? && !argum.nil?
```

Comment:
1- returns false if the block ever returns true for all elements::range
     Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))
 ```
      expected: false
            got: true
```
      note: It was compared using ==;

2- when no block or argument is given returns false only if one of the collection members is true
     Failure/Error: expect(true_array.my_none?).to be false

       expected false
            got true

3- when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
     Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

       expected false
            got true